crates/ruff/src/message/gitlab.rs

```diff
@@ -116,8 +121,7 @@ fn fingerprint(
     let mut hasher = DefaultHasher::new();
 
     kind.rule().hash(&mut hasher);
-    start_location.hash(&mut hasher);
-    end_location.hash(&mut hasher);
+    salt.hash(&mut hasher);
```

Comment:
Nit: I would move the salt to the top (line 123) as I consider it an initialization of the hashing. 

And we can use `hasher.write_u64(salt)` directly after changing salt to a u64